src/main/java/org/folio/support/KafkaEvent.java

```diff
@@ -0,0 +1,72 @@
+package org.folio.support;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.Getter;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.messaging.MessageHeaders;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+@Log4j2
+@Getter
+public class KafkaEvent {
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+  public static final String STATUS = "status";
+  public static final String ITEM_ID = "itemId";
+  private EventType eventType;
+  private JsonNode newNode;
+  private JsonNode oldNode;
+
+  public KafkaEvent(String eventPayload) {
+    try {
+      JsonNode jsonNode = objectMapper.readTree(eventPayload);
+      setEventType(jsonNode.get("type").asText());
+      setNewNode(jsonNode.get("data"));
+      setOldNode(jsonNode.get("data"));
+    } catch (Exception e) {
+      log.error("Could not parse input payload for processing event", e);
+    }
+  }
+
+  private void setEventType(String eventType) {
+    this.eventType = EventType.valueOf(eventType);
+  }
+
+  private void setNewNode(JsonNode dataNode) {
+    if (dataNode != null) {
+      this.newNode = dataNode.get("new");
+    }
+  }
+
+  private void setOldNode(JsonNode dataNode) {
+    if (dataNode != null) {
+      this.oldNode = dataNode.get("old");
+    }
+  }
+
+  public boolean hasNewNode() {
+    return newNode != null;
+  }
+
+  public enum EventType {
+    UPDATED, CREATED
+  }
+
+  public static UUID getUUIDFromNode(JsonNode node, String fieldName) {
+    if (node == null || !node.has(fieldName)) {
+      return null;
+    }
+    return UUID.fromString(node.get(fieldName).asText());
+  }
+
+  public static List<String> getHeaderValue(MessageHeaders headers, String headerName, String defaultValue) {
```

Comment:
This method returns an empty list or a list with 1 string. Can we just return this string?